cmd/kpod/stats.go

```diff
@@ -120,126 +110,117 @@ func getStats(server *libkpod.ContainerServer, args []string, all bool, statsCha
 		step = 0
 	}
 	for i := 0; i < times; i += step {
-		reportStats := []*libkpod.ContainerStats{}
+		reportStats := []*libpod.ContainerStats{}
 		for _, ctr := range ctrs {
 			id := ctr.ID()
 			if _, ok := containerStats[ctr.ID()]; !ok {
-				initialStats, err := server.GetContainerStats(ctr, &libkpod.ContainerStats{})
+				initialStats, err := ctr.GetContainerStats(&libpod.ContainerStats{})
 				if err != nil {
 					return err
 				}
 				containerStats[id] = initialStats
 			}
-			stats, err := server.GetContainerStats(ctr, containerStats[id])
+			stats, err := ctr.GetContainerStats(containerStats[id])
 			if err != nil {
 				return err
 			}
 			// replace the previous measurement with the current one
 			containerStats[id] = stats
 			reportStats = append(reportStats, stats)
 		}
-		statsChan <- reportStats
-
-		err := server.Update()
+		ctrs, err = containerFunc()
 		if err != nil {
 			return err
 		}
-		ctrs, err = server.ListContainers(isRunning, ctrInList(args))
-		if err != nil {
-			return err
+		if strings.ToUpper(format) != "JSON" && !c.Bool("no-reset") {
```

Comment:
There is a const called `JSONString` in formats.go, so you can just do `if format != JSONString`

Reply:
fixed